### ISSUE TITLE
MathML for multi-digit numbers is wrong

### ISSUE THREAD

**Issue by NSoiffer** [Khan/KaTeX#203]
From [this page on trig functions](https://www.khanacademy.org/math/trigonometry/less-basic-trigonometry/trig-symmetry-periodicity/e/symmetry-and-periodicity-of-trig-functions), this is the MathML for the first expr:  sin(α)=0.34

``` html
<math><semantics>
<mrow>
  <mo>sin</mo>
   <mo>(</mo>
   <mi>α</mi>
   <mo>)</mo>
   <mo>=</mo>
   <mn>0</mn><mi mathvariant="normal">.</mi><mn>3</mn><mn>4</mn>
</mrow>
<annotation encoding="application/x-tex">\sin (\alpha) = 0.34</annotation>
</semantics></math>
```

Notice that what should be `<mn>0.34</mn>` has turned into three `<mn>`s and one `<mi>`.

The "sin" should also be in a `<mi>` -- that's [#202](https://github.com/Khan/KaTeX/issues/202)

**Comment by xymostech** [comment]
I noticed this problem when trying out the screenreader, thanks for reporting it.

Unfortunately, there isn't a trivial fix for this. The TeX parser splits strings into individual characters (since TeX has no real conception of semantics), so by the time we're building the MathML tree, there's no information that the two numbers should go together.

I could try doing something like "Combine all adjacent `<mn>[0-9]</mn>` and `<mi>.</mi>` into a single `<mn>`", which would require a little work but wouldn't be awful. It would require a little more work to get `-2.3` to turn into a single `<mn>` (but shouldn't be that bad). However, then something like `01.2.3` would get merged into a single `<mn>`. Does that sound like an acceptable loss? I'm not sure how a screenreader would read `<mn>01.2.3</mn>` (maybe you could try it?)

**Comment by NSoiffer** [comment]
In C++, there are functions to parse strings to numbers and they return the point at which they left off. In JS, it looks like the closest you get is parseInt() and parseFloat(). They don't return where the parse left off, so they are less useful.

I think that the easy thing to do that will be right most of the time is write a simple functions that checks for digits and and decimal points (two separate loops for digits with a check for "." or a flag to say you already saw the decimal point). If you want to worry about internationalization, then you get into using "," as a decimal separator. I also don't know if large numbers are written as "1,234" or even "1 234 567" (i.e, spaces as separators).  Dealing with just digits and "." will go a long way to getting it right.

**Comment by kevinbarabash** [comment]
A similar issue is that we don't collapse multiple primes into a single glyph.  `\u2032` is single prime, `\u2033` is double prime, and `\u2034` is triple prime.  Having a screen reader read out "double prime" or "triple prime" would be much nicer than "prime prime" or "prime prime prime".

**Comment by edemaine** [comment]
Thanks for revitalizing this old issue, @kevinbarabash .  [MDN](https://developer.mozilla.org/en-US/docs/Web/MathML/Element/mn) says "The MathML `<mn>` element represents a numeric literal which is normally a sequence of digits with a possible separator (a dot or a comma)".  So I'd be happy with leaving a `-` (or `+`) outside the `<mn>`, and just concatenating strings of `<mn>` and one (or more) `<mi>.</mi>`s.

We could similarly combine multiple primes, e.g., in `groupTypes.mrow`.  Maybe a good target after #788.

**Comment by kevinbarabash** [comment]
Marking this as an accessibility issue too b/c I'm sure the browser will read "one two three" instead of "one hundred twenty three".
